src/main/java/mate/academy/spring/config/SecurityConfig.java

```diff
@@ -0,0 +1,58 @@
+package mate.academy.spring.config;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.provisioning.InMemoryUserDetailsManager;
+
+@EnableWebSecurity
+@PropertySource("classpath:db.properties")
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final Environment env;
+
+    public SecurityConfig(Environment env) {
+        this.env = env;
+    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+        auth.userDetailsService(inMemoryUserDetailsManager());
+    }
+
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+                .authorizeRequests()
+                .anyRequest().authenticated()
+                .and()
+                .formLogin()
+                .permitAll()
+                .and()
+                .httpBasic()
+                .and()
+                .csrf().disable();
+    }
+
+    @Bean
+    public InMemoryUserDetailsManager inMemoryUserDetailsManager() {
+        List<UserDetails> userDetailsList = new ArrayList<>();
+        userDetailsList.add(User.withUsername(env.getProperty("user.name"))
+                .password(encoder().encode(env.getProperty("user.password")))
+                .roles(env.getProperty("user.role")).build());
+        return new InMemoryUserDetailsManager(userDetailsList);
+    }
+
+    @Bean
+    public PasswordEncoder encoder() {
```

Comment:
method names should start with verbs
```suggestion
    public PasswordEncoder getEncoder() {
```

Reply:
I know it, but here we have bean name, when we use the @Bean annotation on a method, Spring uses the method name as a bean name. So it's quite illogical to call it using verbs.